### PR TITLE
Adds outline property to bs-button

### PR DESCRIPTION
**Pull request by dgavey**
I think this should do it, and it leaves `outline-primary` as still valid. I was a little unsure on the docs on if I set that up correct for you.  Also I could add a test for each style of outline button, but that didn't seem necessary. As usual, leave me any suggestions and I'll address them.

This is as per Issue #673

### CODE REVIEW

**Comment by dgavey** [comment]
Ember 2.16 failed to even build for some reason.  All other tests passed.

File: addon/components/bs4/bs-button.js [hunk 1]
```diff
@@ -1,5 +1,16 @@
 import Button from 'ember-bootstrap/components/base/bs-button';
 
 export default Button.extend({
-  type: 'secondary'
-});
\ No newline at end of file
+  type: 'secondary',
+  classNameBindings: ['outline:btn-outline'],
```
Comment (simonihmig) [inline addon/components/bs4/bs-button.js]: This is setting a `btn-outline` class, right? But is this really needed or used somewhere? The examples [here](http://getbootstrap.com/docs/4.1/components/buttons/#outline-buttons) don't mention this class.

File: tests/integration/components/bs-button-test.js [hunk 2]
```diff
@@ -76,6 +77,11 @@ module('Integration | Component | bs-button', function(hooks) {
     assert.dom('button i').hasClass('fa-check');
   });
 
+  testBS4('button with outline property shows as outline', async function(assert) {
+    await render(hbs`{{bs-button type="primary" outline=true}}`);
+    assert.dom('button').hasClass('btn-outline-primary');
```
Comment (simonihmig) [inline tests/integration/components/bs-button-test.js]: Can you add another assertion here that it does *not* have the `btn-primary` class? (I know it's working right, but just to make sure it)

File: addon/components/bs4/bs-button.js (hunk 1 above)
Comment (dgavey) [reply addon/components/bs4/bs-button.js]: Oops accidentally left in some test code.

File: tests/integration/components/bs-button-test.js (hunk 2 above)
Comment (simonihmig) [reply tests/integration/components/bs-button-test.js]: I meant to add this to the existing test:

```suggestion
    assert.dom('button').hasClass('btn-outline-primary');
    assert.dom('button').doesNotHaveClass('btn-primary');
```

File: tests/integration/components/bs-button-test.js [hunk 3]
```diff
@@ -76,6 +77,16 @@ module('Integration | Component | bs-button', function(hooks) {
...
+  });
+
+  testBS4('button with does not have outline class when outline is null', async function(assert) {
+    await render(hbs`{{bs-button type="primary"}}`);
+    assert.dom('button').doesNotHaveClass('btn-outline-primary');
```
Comment (simonihmig) [inline tests/integration/components/bs-button-test.js]: I don't think we need a separate test for this. If you want you can add this assertion to the existing `'button has correct type'` test.

**Comment by simonihmig** [comment]
> Sorry about the delay in addressing the feedback. Got really sick last week.

@dgavey Nothing to excuse yourself for, hope you're doing well again! 

Thanks for your recent contributions here, appreciate that! 👍

**Pull request merged**